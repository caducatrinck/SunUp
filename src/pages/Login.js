import LogoComponent from '../components/Logo';
import { styled } from 'styled-components';
import CustomButton from '../components/CustomButton';
import { CustomTheme } from '../helpers/CustomTheme';
import React from 'react';
import { useSetRecoilState } from 'recoil';
import { fakeTokenAtom } from '../contexts/States';
import { TextInput, View } from 'react-native';
import CustomText from '../components/CustomText';

export default function Login() {
	const [name, setName] = React.useState('');
	const nameGlobalState = useSetRecoilState(fakeTokenAtom);
	const [error, setError] = React.useState(false);

	React.useEffect(() => {
		const regexOnlyLetters = /^[a-zA-Z\s]+$/;
		if (!regexOnlyLetters.test(name) && name !== '') {
			setError('O nome deve conter apenas letras!');
			return;
		}
		if (name.length > 20) {
			setError('Você atingiu o máximo de caracteres!');
			return;
		}
		setError('');
	}, [name]);

	function setNameAndGoHome() {
		nameGlobalState(name);
	}

	return (
		<BackgrondView>
			<LogoContainer>
				<LogoComponent />
			</LogoContainer>

			<FormContainer>
				<InputContainer>
					<CustomInput
						onChangeText={(e) => setName(e)}
						value={name}
						placeholder="DIGITE SEU NOME"
						keyboardType="default"
					/>
				</InputContainer>
				<CustomButton
					disable={error !== ''}
					onClick={() => setNameAndGoHome()}
					fontSize="20px"
					heigth="30%"
					text="ENTRAR"
				/>
				{error !== '' && (
					<ErroTextContainer>
						<CustomText
							text={error}
							font="upperFont"
							color={CustomTheme.white}
						/>
					</ErroTextContainer>
				)}
			</FormContainer>
		</BackgrondView>
	);
}

const ErroTextContainer = styled(View)`
	height: 20px;
	display: flex;
	justify-content: center;
	align-items: center;
`;
const LogoContainer = styled(View)`
	padding-top: 50px;
	height: 50%;
	display: flex;
	justify-content: center;
`;
const FormContainer = styled(View)`
	height: 50%;
	display: flex;
	padding-top: 40px;
	justify-content: flex-start;
	gap: 10px;
	width: 65%;
	flex-direction: column;
`;
const InputContainer = styled(View)`
	background-color: ${CustomTheme.colors.white};
	border-radius: 5px;
`;
const CustomInput = styled(TextInput)`
	height: 40px;
	padding: 0px 10px 0px 10px;
	font-size: 15px;
	font-style: normal;
	font-family: ${CustomTheme.upperFont};
`;

const BackgrondView = styled(View)`
	width: 100%;
	height: 100%;
	display: flex;
	justify-content: space-evenly;
	align-items: center;
	background-color: ${CustomTheme.colors.background};
	background-color: ${CustomTheme.colors.backgroundLinear};
`;
