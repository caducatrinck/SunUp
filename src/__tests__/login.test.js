import renderer from 'react-test-renderer';
import React from 'react';
import Login from '../pages/Login';
import { cleanup } from '@testing-library/react-native';

afterEach(cleanup);

test('Renderização correta?', () => {
	const tree = renderer.create(<Login />).toJSON();
	expect(tree).toMatchSnapshot();
});
