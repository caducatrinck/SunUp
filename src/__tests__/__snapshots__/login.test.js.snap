// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização correta? 1`] = `
<View
  className="sc-laEPQd iSDLa-d"
>
  <View
    className="sc-kPNBzt dsDssT"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="239.891"
      bbWidth="233.802"
      className="css-1j8o68f"
      focusable={false}
      height={239.891}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 239,
            "width": 233,
          },
        ]
      }
      vbHeight={261.376}
      vbWidth={250}
      width={233.802}
    >
      <RNSVGGroup>
        <RNSVGPath
          d="m100.282 52.19-23.03-16.77c-.375 1.25-.75 2.502-1 3.754l21.902 16.02L75.25 45.056c-.125 1.252-.25 2.503-.25 3.88l21.652 9.637-21.527-4.63c.125 1.25.25 2.628.376 3.88l20.4 4.38H76.377c.25 1 .626 2.253 1.001 3.504h18.523l-17.146 3.63 1.502 3.379 16.395-3.505L82.01 75.72c.626 1 1.251 2.127 2.002 3.128l14.143-6.257-12.265 8.886a25.027 25.027 0 0 0 2.378 2.753l12.015-8.76-9.888 10.888a25.027 25.027 0 0 0 2.754 2.378l9.887-10.889-7.384 12.766c1.001.751 2.002 1.377 3.004 2.003l7.509-13.017-4.756 14.644a39.43 39.43 0 0 0 3.254 1.627l4.881-15.144-1.752 16.52c1.126.376 2.378.751 3.504 1.127l1.753-17.272 1.877 18.148c1.251.25 2.503.5 3.755.626l-2.003-19.15 6.383 19.525h3.755L120.18 79.6l11.64 20.15c1.252-.126 2.503-.376 3.755-.626L123.31 77.722l17.773 19.775c1.251-.376 2.378-.876 3.63-1.377L125.938 75.22l24.655 17.897a29.115 29.115 0 0 0 3.255-2.127l-25.783-18.774 31.415 14.018c.876-.876 1.877-1.752 2.753-2.754l-32.666-14.518 37.672 8.01c.626-1.126 1.377-2.128 1.878-3.254l-38.924-8.26h42.428c.375-1.127.751-2.378 1.001-3.63h-43.429L175 52.315v-3.754l-45.557 9.637 44.305-19.775c-.25-1.252-.625-2.378-1-3.504L127.94 54.944l40.676-29.537c-.626-1.002-1.252-2.128-1.877-3.13l-40.926 29.788 34.167-37.922c-.876-.876-1.752-1.627-2.628-2.378l-34.168 37.922 25.282-43.93c-1.001-.626-2.128-1.126-3.254-1.627l-25.157 43.68L135.326.876c-1.127-.25-2.379-.5-3.63-.626l-15.019 46.433L121.558 0c-1.251.125-2.503.25-3.63.375l-4.88 45.933-4.631-43.68a40.583 40.583 0 0 0-3.505 1.377l4.506 42.678-12.39-38.298c-1.002.751-2.128 1.502-3.13 2.253l12.016 37.172-18.148-31.415a63.596 63.596 0 0 0-2.503 3.004l17.522 30.288-21.652-24.03c-.626 1.126-1.252 2.253-1.753 3.504l20.902 23.03zm9.01 3.63c1.127-1.127 3.005-1.127 4.131 0 1.126 1.126 1.126 3.003 0 4.13-1.126 1.126-3.004 1.126-4.13 0-1.127-1.127-1.127-3.004 0-4.13zm-3.504 16.145a6.723 6.723 0 0 1 0-9.512l9.512 9.512a6.723 6.723 0 0 1-9.512 0zm15.77-3.88c-1.126 1.126-3.004 1.126-4.13 0s-1.126-3.004 0-4.13c1.126-1.127 3.004-1.127 4.13 0 1.127 1.126 1.127 3.004 0 4.13z"
          fill={
            {
              "payload": 4294940928,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M0 195.266h13.43c.267 17.154 1.597 20.479 6.384 20.479 6.516 0 6.516-5.452 6.516-19.947C26.33 174.52 0 180.239 0 151.915 0 135.293 0 120 19.947 120c14.76 0 18.75 8.378 19.68 28.457H26.33c-.532-14.494-2.128-17.819-6.383-17.819-6.516 0-6.516 7.181-6.516 21.277 0 20.612 26.463 13.298 26.463 43.883 0 18.617 0 33.245-20.08 33.245-15.426 0-19.016-9.176-19.814-33.777zm49.867-23.537v-50.4h13.298v50.4c0 38.43 0 44.016 6.649 44.016s6.649-5.585 6.649-44.016v-50.4H89.76v50.4c0 42.553 0 57.314-19.947 57.314s-19.947-14.761-19.947-57.314zm52.527 55.984V121.33h11.968l17.42 65.16-.133-1.33v-63.83h13.298v106.383h-11.968l-17.42-58.51.132 1.329v57.18h-13.297zm55.186-55.984v-50.4h13.298v50.4c0 38.43 0 44.016 6.649 44.016s6.649-5.585 6.649-44.016v-50.4h13.297v50.4c0 42.553 0 57.314-19.946 57.314s-19.947-14.761-19.947-57.314zm52.526 55.984V121.33h21.543c13.697 0 17.287 8.377 18.351 28.457v4.256c-1.064 20.08-4.654 28.457-18.351 28.457h-8.245v45.213h-13.298zm13.298-55.851h8.245c3.99 0 5.452-3.325 5.984-17.82v-4.255c-.532-14.494-1.995-17.819-5.984-17.819h-8.245v39.894z"
          fill={
            {
              "payload": 4293848814,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M25.024 258.024h-.017c0 .407-.016.83 0 1.253.017.52.049 1.041.082 1.562 0 .179.048.228.211.26.342.082.667.098 1.009.098.879-.033 1.757.032 2.636 0 .358-.016.52.016.879-.049.325-.065.44-.244.44-.732-.033-.31-.017-.44-.017-.814 0-.537-.374-.716-.7-.7-.228.017-1.318-.016-1.757-.016-.732.016-.619.114-.619-.65 0-.619.017-1.416.017-2.051 0-.342.032-.456.358-.456h2.034c.325.05.553-.179.553-.52s.032-.358.032-.7l-.048-.488c-.033-.277-.13-.374-.407-.374-.18 0-.358.016-.537.032-.7.049-1.025-.049-1.709 0-.227 0-.26-.032-.276-.244-.065-.635 0-1.334.065-1.969.016-.26.081-.31.325-.325l2.295-.017c.309-.032.406-.016.39-.325-.032-.472.016-.96-.032-1.432-.033-.26-.114-.31-.407-.31-.488.017-.602.017-1.09.05-.489.016-2.441.016-2.913.016-.684 0-.732-.049-.749.781-.032.65-.016 1.318-.016 1.985 0 1.14-.065 5.224-.032 6.135zm15.61-8.494c.048.228.016 11.065.016 11.341 0 .489-.18.472-.505.472-.488 0-1.171.082-1.318-.081-.13-.114-2.538-5.923-2.88-6.655-.033-.065-.114-.228-.179-.196-.081.033-.049.147-.049.212-.016.57.082 5.484.082 6.265 0 .39-.017.455-.342.455h-.814c-.211 0-.472-.016-.683.017-.163 0-.244-.05-.26-.212v-1.285c.032-1.367 0-9.47 0-10.073 0-.52.146-.618.667-.618.358-.017.716-.017 1.074-.017.113 0 .179.017.244.163.439 1.025 2.31 5.728 2.603 6.298.049.113.228.569.326.618h.016c.065-.065.049-.39.049-.488 0-.244-.13-5.956-.13-6.07 0-.163-.05-.309.032-.423.049-.065.26-.081.342-.098.455-.032.992.017 1.448 0 .228 0 .212.163.26.375zm3.486 8.494h-.016c0 .407-.016.83 0 1.253.016.52.049 1.041.082 1.562 0 .179.048.228.211.26.342.082.667.098 1.009.098.879-.033 1.757.032 2.636 0 .358-.016.52.016.879-.049.325-.065.44-.244.44-.732-.033-.31-.017-.44-.017-.814 0-.537-.374-.716-.7-.7-.228.017-1.318-.016-1.757-.016-.733.016-.619.114-.619-.65 0-.619.017-1.416.017-2.051 0-.342.032-.456.358-.456h2.034c.325.05.553-.179.553-.52s.032-.358.032-.7l-.048-.488c-.033-.277-.13-.374-.407-.374-.18 0-.358.016-.537.032-.7.049-1.025-.049-1.709 0-.228 0-.26-.032-.276-.244-.065-.635 0-1.334.065-1.969.016-.26.081-.31.325-.325l2.295-.017c.309-.032.406-.016.39-.325-.032-.472.016-.96-.032-1.432-.033-.26-.114-.31-.407-.31-.488.017-.602.017-1.09.05-.489.016-2.441.016-2.913.016-.684 0-.732-.049-.749.781-.032.65-.016 1.318-.016 1.985 0 1.14-.065 5.224-.033 6.135zm10.696 2.115c0-.423 0-3.092.016-3.303.017-.049.05-.147.114-.147s.098.033.114.082c.651 1.27 1.22 2.538 1.806 3.775.082.179.26.7.31.748.065.098.179.066.276.066l1.79.016c.212 0 .26-.13.212-.31-.13-.487-1.058-2.212-1.27-2.635-.146-.326-1.122-2.213-1.155-2.36-.032-.065.212-.179.244-.228.781-.7 1.481-1.415 1.627-2.506.163-1.074-.08-2.245-.78-3.108-.538-.634-1.237-1.008-2.067-1.057-.505-.033-.993.016-1.497 0-.44-.017-.749-.017-1.302-.017-.26 0-.423.147-.44.424v.195c-.032 1.253 0 2.506.017 3.742 0 1.888-.033 6.867-.033 7.583 0 .212.033.26.228.26.114 0 .244.017.358.017.456 0 .732-.016 1.204-.016.358 0 .228-.586.228-1.22zm0-6.346.016-1.643c.017-.245-.065-.733.163-.912.31-.228.944 0 1.22.196.814.553.75 1.92.31 2.684-.212.358-.57.57-.977.684-.244.065-.7.146-.716-.244 0-.26-.016-.765-.016-.765zm15.366 4.833c.016.78 0 1.578-.049 2.36 0 .064-.049.13-.098.162a.389.389 0 0 1-.227.081c-1.01.163-2.23.244-3.206-.081-2.197-.748-3.694-3.206-3.694-6.02 0-3.044 2.18-6.005 5.354-6.119.423-.016 1.871-.081 1.822.635-.032.52.017.943 0 1.48 0 .098-.065.163-.162.147-.928-.114-1.986-.407-2.88.016-.75.358-1.27 1.123-1.563 1.871-.52 1.237-.407 2.669-.049 3.938.277 1.01.88 2.197 2.083 2.23.065.016.684.048.684-.017l-.05-2.083c-.015-.097-.015-.195-.032-.292s-.081-.13-.179-.147c-.26 0-.26-.016-.52-.016-.602 0-.472-.407-.489-.537-.016-.244 0-.472-.016-.716-.016-.228.049-.407.277-.44h.195c.846-.048 1.709.017 2.604.017.13 0 .195.163.195.309.016.163.016.277.016.44.049.797-.049 1.61-.032 2.408.016.13.016.244.016.374zm5.716-1.465c-.048-1.855-.032-3.824-.016-5.662 0-.57-.032-1.188-.032-1.758 0-.618-.245-.586-.57-.602-.472-.016-.683-.016-1.155-.016-.31-.016-.44.065-.44.374 0 1.595-.049 6.102-.032 6.753s.081 3.613.049 4.263c-.017.228 0 .456.016.684.016.081.065.163.195.163.537-.033 1.042-.033 1.579-.05.195-.016.358-.064.439-.276.016-.114-.033-3.092-.033-3.873zm4.968 4.068c.049.017.195.017.342.017.325 0 .423-1.172.456-1.416.016-.195.065-.212.227-.244.814-.098 1.22-.033 2.034-.016.18 0 .277.081.326.227l.114.375c.097.276.179.618.276.895.065.195.082.179.277.179.26 0 1.204 0 1.48-.017.375 0 .115-.764.066-.895-.147-.455-.18-.927-.326-1.383-.065-.195-1.92-8.998-2.001-9.486-.114-.293-.016-.31-.277-.31-.553 0-1.139.033-1.708 0-.228 0-.212.066-.244.31-.033.227-.619 2.815-.847 3.889-.292 1.35-.553 2.652-.878 3.986-.114.472-.895 3.873-.798 3.89.05.032.13.016.196.016.407 0 .78 0 1.285-.017zm2.132-3.693h-.342c-.455 0-.472 0-.358-.472.082-.358.52-2.295.618-2.767.017-.032.05-.065.066-.065.032 0 .065.033.08.05.066.194.131.683.196.878.147.618.244 1.139.39 1.806.147.586.082.586-.21.586-.147 0-.294-.016-.44-.016zm12.913-6.786c-.016 1.009.082 8.38.066 8.917 0 .505 0 1.01.016 1.514 0 .08.049.162.13.162 1.139.05 2.262-.049 3.4-.032.375.016.75 0 1.107 0 .358-.017.39-.163.375-.456-.017-.455-.017-.862-.017-1.302-.016-.227-.146-.309-.341-.309-.668.016-1.335-.016-2.002.016a2.403 2.403 0 0 1-.342.017c-.195-.017-.244-.065-.26-.277v-.244c-.049-1.35.049-2.75.049-4.1-.016-1.14 0-2.165-.016-3.174 0-.439 0-.878-.017-1.301-.016-.26 0-.537-.065-.798-.032-.097-.081-.228-.195-.228h-1.53c-.26 0-.341.18-.358.44v1.155zm10.777 6.411c-.048-1.855-.032-3.824-.016-5.662 0-.57-.032-1.188-.032-1.758 0-.618-.244-.586-.57-.602-.472-.016-.683-.016-1.155-.016-.31-.016-.44.065-.44.374 0 1.595-.048 6.102-.032 6.753s.081 3.613.049 4.263c-.017.228 0 .456.016.684.016.081.065.163.195.163.537-.033 1.042-.033 1.579-.05.195-.016.358-.064.439-.276.016-.114-.033-3.092-.033-3.873zm5.83-7.99c-.162 0-.325.017-.487.017-.244 0-.537-.049-.586.277-.163 1.383-.31 2.912-.472 4.295-.277 2.279-.537 4.622-.797 6.9-.033.309-.066.602.276.667.52.033.911.033 1.35.016.164.017.342.033.391-.162.033-.082.033-.163.033-.245.032-.374.52-5.06.716-5.06.309 0 1.123 3.808 1.513 5.207.049.195.147.277.342.26.407-.032.797 0 1.204 0s.325-.26.407-.602c.097-.374 1.041-4.881 1.269-4.881.26 0 .797 5.223.846 5.337.098.228.374.13.57.13.293 0 .57.033.943 0 .277-.033.554.065.57-.293 0-.081-.033-.163-.049-.244a36.324 36.324 0 0 0-.228-1.302c-.114-.846-.276-1.676-.39-2.522-.163-1.334-.326-2.669-.472-4.003-.114-1.139-.277-2.424-.374-3.563-.033-.375-.456-.245-.716-.245h-1.123c-.179 0-.26.342-.293.521-.098.456-1.22 6.33-1.595 6.33s-1.578-5.37-1.79-6.541c-.048-.26-.227-.342-.471-.342zm11.543 2.116c.016.96.016 9.113.016 9.78 0 .244.032.293.276.293.554 0 .554.016 1.107 0 .114 0 .31 0 .407-.05.114-.048.049-3.888.049-4.1 0-.228.455-.179.585-.179 2.718 0 3.466-1.952 3.466-3.612 0-1.497-.293-4.264-3.677-4.264-.586 0-1.188.017-1.758.017-.13 0-.292-.017-.406.065-.13.097-.082 1.546-.065 2.05zm1.87 3.352c0-.358-.015-2.49 0-3.14 0-.098-.015-.212 0-.326.05-.179.261-.179.424-.163.944.066 1.48.895 1.562 1.774.065.635-.016 1.579-.472 2.083a1.726 1.726 0 0 1-1.302.553c-.113 0-.227-.13-.227-.244 0-.179 0-.358.016-.537zm8.972 6.59c.049.017.195.017.341.017.326 0 .423-1.172.456-1.416.016-.195.065-.212.228-.244.813-.098 1.22-.033 2.034-.016.179 0 .276.081.325.227l.114.375c.098.276.18.618.277.895.065.195.081.179.276.179.26 0 1.205 0 1.481-.017.374 0 .114-.764.065-.895-.146-.455-.179-.927-.325-1.383-.065-.195-1.92-8.998-2.002-9.486-.114-.293-.016-.31-.276-.31-.553 0-1.14.033-1.709 0-.228 0-.211.066-.244.31-.032.227-.618 2.815-.846 3.889-.293 1.35-.553 2.652-.879 3.986-.114.472-.895 3.873-.797 3.89.049.032.13.016.195.016.407 0 .781 0 1.286-.017zm2.131-3.693h-.341c-.456 0-.472 0-.358-.472.081-.358.52-2.295.618-2.767.016-.032.049-.065.065-.065.033 0 .065.033.081.05.066.194.13.683.196.878.146.618.244 1.139.39 1.806.147.586.082.586-.211.586-.147 0-.293-.016-.44-.016zm14.98 2.05c.017.504.017 1.009.033 1.513 0 .228.016.26.244.26h1.465c.309 0 .244-.52.227-.797l.05-9.112c0-.244.08-.326.406-.326h1.432c.342 0 .26-.585.26-.992 0-.358.13-.977-.26-.977-.277 0-2.929 0-4.02.017-.536 0-.976-.017-1.513-.017-.406 0-.39.326-.39 1.042 0 .456-.016.911.374.911.228 0 1.155 0 1.416.016.163 0 .276.05.309.228 0 .049.016.098.016.147 0 1.171-.081 6.622-.049 8.087zm16.147-7.436c.342 1.008.423 2.099.423 3.173 0 .895-.162 1.757-.39 2.62-.293 1.025-.781 1.952-1.66 2.554-.65.472-1.334.781-2.132.863-.113 0-.211.016-.325.016-.098 0-.195-.016-.31-.016-.797-.082-1.48-.391-2.147-.863-.863-.602-1.367-1.53-1.644-2.555-.244-.862-.39-1.724-.39-2.62 0-1.073.081-2.164.423-3.172.602-1.742 2.262-2.978 4.068-3.01 1.806.032 3.466 1.268 4.084 3.01zm-1.578 4.36c.163-.992.098-2.001-.098-2.994-.195-.895-.586-1.773-1.334-2.31a1.527 1.527 0 0 0-.423-.228c-.212-.098-.44-.13-.651-.13s-.423.032-.635.13c-.162.049-.293.13-.423.228-.764.537-1.155 1.415-1.334 2.31-.195.993-.26 2.002-.098 2.994.163.895.489 2.1 1.335 2.555.358.195.748.326 1.155.342a2.711 2.711 0 0 0 1.172-.342c.846-.455 1.171-1.66 1.334-2.555zm12.29 1.855c.098-.211.18-.423.26-.634.18-.52.212-1.058.261-1.595.098-1.139.065-2.294-.228-3.4-.358-1.416-1.27-2.718-2.636-3.32-.618-.277-1.318-.31-1.985-.293-.7 0-1.416-.033-2.116.032-.195.017-.292.114-.325.293-.033.196-.065.39-.065.57.032.992 0 3.938.032 4.426.017.44-.032.976-.032 1.416.016 1.09 0 2.196 0 3.287 0 .455-.016.895-.016 1.35 0 .488-.18.7 1.155.7 1.188 0 2.425-.098 3.547-.52 1.042-.375 1.693-1.335 2.148-2.312zm-4.735-1.692v-1.952c0-1.237.033-2.034-.016-3.076v-.195c.016-.342.065-.358.423-.358.472-.016.813.097 1.22.26s.635.52.846.863c.407.667.57 1.399.602 2.164.017.44.033.879.033 1.318 0 2.083-1.22 3.53-2.294 3.53-.716 0-.83.05-.83-.422 0-.488.032-.976.032-1.465-.016-.211-.016-.439-.016-.667zm17.318-4.523c.342 1.008.424 2.099.424 3.173 0 .895-.163 1.757-.391 2.62-.293 1.025-.781 1.952-1.66 2.554-.65.472-1.334.781-2.131.863-.114 0-.212.016-.326.016-.097 0-.195-.016-.309-.016-.797-.082-1.48-.391-2.148-.863-.862-.602-1.367-1.53-1.643-2.555-.244-.862-.39-1.724-.39-2.62 0-1.073.08-2.164.422-3.172.602-1.742 2.262-2.978 4.068-3.01 1.806.032 3.466 1.268 4.084 3.01zm-1.578 4.36c.163-.992.098-2.001-.098-2.994-.195-.895-.585-1.773-1.334-2.31a1.527 1.527 0 0 0-.423-.228c-.211-.098-.44-.13-.65-.13s-.424.032-.635.13c-.163.049-.293.13-.424.228-.764.537-1.155 1.415-1.334 2.31-.195.993-.26 2.002-.097 2.994.162.895.488 2.1 1.334 2.555.358.195.748.326 1.155.342a2.711 2.711 0 0 0 1.172-.342c.846-.455 1.171-1.66 1.334-2.555zm18.17 1.855c.097-.211.179-.423.26-.634.179-.52.211-1.058.26-1.595.098-1.139.065-2.294-.228-3.4-.358-1.416-1.269-2.718-2.636-3.32-.618-.277-1.318-.31-1.985-.293-.7 0-1.416-.033-2.115.032-.195.017-.293.114-.326.293-.032.196-.065.39-.065.57.033.992 0 3.938.033 4.426.016.44-.033.976-.033 1.416.017 1.09 0 2.196 0 3.287 0 .455-.016.895-.016 1.35 0 .488-.179.7 1.155.7 1.188 0 2.425-.098 3.548-.52 1.041-.375 1.692-1.335 2.148-2.312zm-4.736-1.692v-1.952c0-1.237.033-2.034-.016-3.076v-.195c.016-.342.065-.358.423-.358.472-.016.814.097 1.22.26s.635.52.847.863c.407.667.57 1.399.602 2.164.016.44.032.879.032 1.318 0 2.083-1.22 3.53-2.294 3.53-.716 0-.83.05-.83-.422 0-.488.033-.976.033-1.465-.017-.211-.017-.439-.017-.667zm10.989.488c-.049-1.855-.033-3.824-.016-5.662 0-.57-.033-1.188-.033-1.758 0-.618-.244-.586-.57-.602-.471-.016-.683-.016-1.155-.016-.309-.016-.44.065-.44.374 0 1.595-.048 6.102-.032 6.753s.082 3.613.05 4.263c-.017.228 0 .456.015.684.017.081.066.163.196.163.537-.033 1.041-.033 1.578-.05.195-.016.358-.064.44-.276.016-.114-.033-3.092-.033-3.873zm4.968 4.068c.049.017.195.017.342.017.325 0 .423-1.172.455-1.416.016-.195.065-.212.228-.244.814-.098 1.22-.033 2.034-.016.179 0 .277.081.325.227l.114.375c.098.276.18.618.277.895.065.195.081.179.277.179.26 0 1.204 0 1.48-.017.375 0 .114-.764.065-.895-.146-.455-.179-.927-.325-1.383-.065-.195-1.92-8.998-2.001-9.486-.114-.293-.017-.31-.277-.31-.553 0-1.14.033-1.709 0-.227 0-.211.066-.244.31-.032.227-.618 2.815-.846 3.889-.293 1.35-.553 2.652-.879 3.986-.114.472-.895 3.873-.797 3.89.049.032.13.016.195.016.407 0 .781 0 1.286-.017zm2.131-3.693h-.341c-.456 0-.472 0-.358-.472.081-.358.52-2.295.618-2.767.016-.032.049-.065.065-.065.033 0 .065.033.082.05.065.194.13.683.195.878.146.618.244 1.139.39 1.806.147.586.082.586-.211.586-.147 0-.293-.016-.44-.016z"
          fill={
            {
              "payload": 4294940928,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    className="sc-bXxmBy fDAUbi"
  >
    <View
      className="sc-jlOLRj dxEMEj"
    >
      <TextInput
        className="sc-dMnqAs hyqSdd"
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="DIGITE SEU NOME"
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="sc-hfDuwB cSjEsD"
        fontSize="20px"
      >
        ENTRAR
      </Text>
    </View>
    <View
      className="sc-dBwcjI jpsVOW"
    >
      <Text
        className="sc-fBEEOr eUYKtO"
        font="upperFont"
        fontSize="15px"
      />
    </View>
  </View>
</View>
`;
